drafts/0007-RFC-BAG-OPERATORS.md

```diff
@@ -44,279 +44,288 @@ INTERSECT(A, B) = DISTINCT(INTERSECT_ALL(A, B))
-]
----
-OK!
+>>
+-- OK!
+
+-- Result Column Descriptors
+-- | Ordinal | Name  | Type   | Nullable |
+-- |---------|-------|--------|----------|
+-- | 0       | _0    | String | NO       |
+-- | 1       | _1    | String | NO       |
 ```
 
-**Union 4** — Combining heterogenous collections
+Notice that we have some issues with the meaning of our keys/columns. For the first three structs/tuples, key *_0* contains a name and *_1* contains a title; yet it is the opposite for the last four structs/tuples. We can fix this by using the SQL *CORRESPONDING* clause.
+
+```sql
+SELECT * FROM hr.employees
+UNION ALL CORRESPONDING BY ( name, title )
+SELECT * FROM engineering.employees
+
+-- Equivalent to *Union 1*
 
+SELECT name, title FROM hr.employees
+UNION ALL
+SELECT name, title FROM engineering.employees
 ```
-{
-  'colors': {
-    'rgb': <<
-      { 'red': [ 255, 0, 0 ] },
-      { 'green': [ 0, 255, 0 ] },
-      { 'blue': [ 0, 0, 255 ] },
-      { 'black': [ 0, 0, 0 ] }
-    >>,
-    'cmyk': <<
-      { 'name': 'cyan', 'CMYK': [ 1, 0, 0, 0 ] },
-      { 'name': 'magenta', 'CMYK': [ 0, 1, 0, 0 ] },
-      { 'name': 'yellow', 'CMYK': [ 0, 0, 1, 0 ] },
-      { 'name': 'black', 'CMYK': [ 0, 0, 0, 1 ] }
-    >>
-  }
-}
+
+**Union 3** — Heterogenous Bags with Strict Typing
+
+Here we attempt to union the job titles in HR and Engineering, but there is a mistake. In the absence of schema, the bags produced by *T1* and *T2* are unioned as two bags of structs, but in strict-typing mode, we need to look at the column descriptors. This behavior conforms to the SQL specification.
+
+```sql
+SELECT * FROM hr.employees               -- T1
+UNION
+SELECT title FROM engineering.employees  -- T2
+-- ERROR!
+-- T1 and T2 are not union-compatible in strict-typing mode
 ```
 
 ```
--- RGB values of all colors
-SELECT * FROM colors.rgb
-UNION ALL
-PIVOT [
-  255*(1-c.CMYK[0])*(1-c.CMYK[3]),
-  255*(1-c.CMYK[1])*(1-c.CMYK[3]),
-  255*(1-c.CMYK[2])*(1-c.CMYK[3])
-] AT c.name FROM colors.cmyk as c
-
-PartiQL> SELECT * FROM colors.rgb
-   | UNION ALL
-   | PIVOT [
-   |   255*(1-c.CMYK[0])*(1-c.CMYK[3]),
-   |   255*(1-c.CMYK[1])*(1-c.CMYK[3]),
-   |   255*(1-c.CMYK[2])*(1-c.CMYK[3])
-   | ] AT c.name FROM colors.cmyk as c
-   |
-==='
+       Column Descriptors of T1                          Column Descriptors of T2 
+
+| Ordinal | Name  | Type   | Nullable |           | Ordinal | Name  | Type   | Nullable |       
+|---------|-------|--------|----------|           |---------|-------|--------|----------|
+| 1       | NAME  | String | NO       |  >UNION<  | 0       | TITLE | String | NO       |
+| 2       | TITLE | String | NO       |
+```
+
+**Union 4** — Heterogenous Bag with Permissive Typing
+
+Let's do a simple union that would cause confusion in strict mode (**Union 2**), but is no problem for permissive mode. Without schemas, the union operators is a pure multi-set union. This shows how PartiQL's UNION operator's functionality is a superset of the SQL UNION operator.
```

Comment:
I think I need an explanation of what is going on here--I would expect this to work in both cases the same.